### PR TITLE
General Housekeeping

### LINKED ISSUE #99: Minor Unit Details Page Updates
# Overview

Some minor updates to the Unit Details page

[Figma](https://www.figma.com/design/06vEDN5gtlHlC5hxNXMb4g/USHS-Designs?node-id=2999-14560&m=dev)

## Frontend

1. Format the landlord phone number nicely in (xxx) xxx-xxxx format (Note that the phone number field is a just a text input on the landlord form, so the phone number can be stored as anything e.g. xxx-xxx-xxxx, xxxxxxxxxx, (xxx)xxxxxxx, etc)
2. Add separating horizontal lines between the sections as in designs
3. Make the address open Google Maps with that address in a new tab
4. Change HL/RS dropdown (in Referrals table) icon from search to arrow (match the arrow of the Status dropdown)

On Home Page:
1. Move "Add Listing" button to its new place.

## Backend

1. When units are deleted, delete all referrals associated with that unit as well.

## Acceptance Criteria
- [x] formatted phone number
- [x] section separators 
- [x] maps link
- [x] referral table icon change
- [x] cacade delete referrals when unit is deleted
- [x] "Add Listing" moved to new home

### PR DESCRIPTION
**Pull request by Azhou2023**
## Tracking Info

Resolves #99 

## Changes

- Formatted phone number in unit details page to nice (xxx) xxx-xxxx format
- Added section separators in unit details page and reformatted some text
- Made address clickable in unit details page which leads to google maps page for it
- Changed UserDropdown icon to up/down arrow for referral table
- Moved "Add Listing" button to top of the page
- Implemented deletion of all unit referrals when an unit is deleted

## Testing

Thoroughly tested all frontend elements and made sure they looked like what was specified by Figma. Tried a bunch of different addresses and made sure they all loaded in google maps. Tested referral deletion with multiple units and ensured that it deleted all the relevant referrals and nothing else.

## Confirmation of Change

Screenshot 2024-05-26 at 6 07 14 PM
Screenshot 2024-05-26 at 6 07 24 PM
Screenshot 2024-05-26 at 6 07 57 PM
Screenshot 2024-05-26 at 6 08 15 PM

### CODE REVIEW

File: frontend/src/pages/UnitDetails.tsx [hunk 1]
```diff
@@ -116,17 +110,24 @@ const StrongText = styled(Text)`
...
-const Address = styled(Header)`
+const Address = styled.a`
+  font-size: 32px;
+  font-weight: 700;
+  font-family: "Neutraface Text";
+  line-height: 150%;
+  line-spacing: 0.64px;
   padding: 0;
   margin: 0 0 5px 0;
+  // text-decoration: none;
```
Comment (PoliteUnicorn) [inline frontend/src/pages/UnitDetails.tsx]: Maybe delete this?

File: frontend/src/components/FilterDropdown.tsx [hunk 2]
```diff
@@ -150,11 +171,23 @@ export const FilterDropdown = (props: FilterDropdownProps) => {
...
+            <span>Listing</span>
+          </AddListings>
+        )}
 
         <ResetFilterButton onClick={resetFilters}>
           <ResetFilterRow>
+            {/* Commented out for now since I'm not sure if we're still using this
             <ResetIcon src="/refresh.svg" />
-            <ResetFilterText> Reset filters</ResetFilterText>
+            <ResetFilterText> Reset filters</ResetFilterText> */}
```
Comment (petabite) [inline frontend/src/components/FilterDropdown.tsx]: Looks like this gets deleted in #101 ([here](https://github.com/TritonSE/USHS-Housing-Portal/pull/101/files#diff-1ea388ab39d9e9f774f6bcffbd624c47e5301c4a86ad6e1d57b4e7de218d5d2cL153)). If you want, we can just leave these uncommented to save us from a merge conflict down the road.

**Review by petabite — APPROVED** [review]
LGTM :ship:

**Review by PoliteUnicorn — APPROVED** [review]
I think this looks good! Thanks for making the delete referral route :D

**Pull request merged**